src/lib/dnssd/Advertiser.h

```diff
@@ -55,6 +55,14 @@ enum class ICDModeAdvertise : uint8_t
     kLIT,  // The ICD is currently operating as a LIT. ICD=1 in DNS-SD key/value pairs.
 };
 
+enum class TCPModeAdvertise : uint8_t
+{
+    kNone            = 0,                         // The device does not support TCP.
+    kTCPClient       = 1 << 1,                    // The device supports the TCP client.
+    kTCPServer       = 1 << 2,                    // The device supports the TCP server.
+    kTCPClientServer = (kTCPClient | kTCPServer), // The device supports both the TCP client and server.
```

Comment:
why not using a BitFlags for storage and have the enums for client/server only?

I am concerned about constants  for all combinations - looks neat when 2 options exist, however once we go to 3 or more options these custom constants will explode.